File: AllGates.py
from NorGate import NorGate
from andGate import AND_GATE
from OrGate import OrGate
from NandGate import NandGate
from notGate import NotGate

nor = NorGate()
nand = NandGate()
o = OrGate()
ng = NotGate()
a = AND_GATE()

ALL_GATES = {
    str(type(nor)): NorGate,
    str(type(nand)): NandGate,
    str(type(o)): OrGate,
    str(type(ng)): NotGate,
    str(type(a)): AND_GATE
}

File: javascriptPort/src/library/AllGates.ts
import AND_GATE from "./AndGate"
import OrGate from "./OrGate"
import NandGate from "./NandGate"
import NotGate from "./NotGate"

const a = new AND_GATE();
const o = new OrGate();
const n = new NandGate();
const no = new NotGate();

export default {
    "AndGate": AND_GATE,
    "OrGate": OrGate,
    "NandGate": NandGate,
    "NotGate": NotGate
}